tests/test_comparison_lib.py

```diff
@@ -59,6 +59,7 @@ def test_distance_function_comparison():
             assert sum(df_pred[f"gamma_{col}"] == gamma_val) == expected_count
 
 
+@mark_with_dialects_excluding()
```

Comment:
If you want to make this backend agnostic, you need to add some extra goodies.

See the construction of the following test - https://github.com/moj-analytical-services/splink/blob/c60e25347caa6cc820032f61444f2ddcecf33984/tests/test_comparison_level_lib.py#L52.

For the backend agnostic logic to work, you need to use pick out a specific backend helper, grab the relevant comparison library (cl) and also supplies `**helper.extra_linker_args()` to the linker object:
```py
linker = helper.Linker(df, settings, **helper.extra_linker_args())
```

If you want to do this, I'd recommend doing it in a separate PR. Then you won't need to worry about rebasing from master.

Reply:
A thorough breakdown can be found here - https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/testing.html#writing-tests.